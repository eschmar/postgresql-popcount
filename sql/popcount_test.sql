CREATE EXTENSION popcount;

SELECT popcount(15::bit(4));
SELECT popcount(7::bit(4));

SELECT popcount(15::bit(8));
SELECT popcount(7::bit(8));

SELECT popcount(15::bit(12));
SELECT popcount(7::bit(12));

SELECT popcount(15::bit(32));
SELECT popcount(7::bit(32));

SELECT popcount(30::bit(32));
SELECT popcount(127::bit(32));

SELECT sum(popcount(x::bit(512))) from generate_series(1,1000000) x;
SELECT sum(popcount(x::bit(500))) from generate_series(1,1000000) x;
