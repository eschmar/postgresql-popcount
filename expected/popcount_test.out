CREATE EXTENSION popcount;
SELECT popcount(15::bit(4));
 popcount 
----------
        4
(1 row)

SELECT popcount(7::bit(4));
 popcount 
----------
        3
(1 row)

SELECT popcount(15::bit(8));
 popcount 
----------
        4
(1 row)

SELECT popcount(7::bit(8));
 popcount 
----------
        3
(1 row)

SELECT popcount(15::bit(12));
 popcount 
----------
        4
(1 row)

SELECT popcount(7::bit(12));
 popcount 
----------
        3
(1 row)

SELECT popcount(15::bit(32));
 popcount 
----------
        4
(1 row)

SELECT popcount(7::bit(32));
 popcount 
----------
        3
(1 row)

SELECT popcount(30::bit(32));
 popcount 
----------
        4
(1 row)

SELECT popcount(127::bit(32));
 popcount 
----------
        7
(1 row)

SELECT popcount(b'10101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010101010101010110101010101010101010101010101010101101010101010101010101010101010101011010101010101010101010');
 popcount 
----------
      263
(1 row)

SELECT sum(popcount(x::bit(512))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

SELECT sum(popcount(x::bit(500))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

